OpenRA.Mods.Common/Traits/World/WarheadDebugOverlay.cs

```diff
@@ -0,0 +1,85 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Part of the combat overlay from DeveloperMode. Attach this to the world actor.")]
+	public class WarheadDebugOverlayInfo : ITraitInfo
+	{
+		public readonly int DisplayDuration = 25;
+
+		public object Create(ActorInitializer init) { return new WarheadDebugOverlay(this); }
+	}
+
+	public class WarheadDebugOverlay : IPostRender
+	{
+		class WHImpact
+		{
+			public readonly WPos CenterPosition;
+			public readonly WDist[] Range;
+			public readonly WDist OuterRange;
+			public int Time;
+
+			public WHImpact(WPos pos, WDist[] range, int time)
+			{
+				CenterPosition = pos;
+				Range = range;
+				Time = time;
+
+				OuterRange = range[range.Length - 1];
```

Comment:
Is there a particular reason to store another copy of the outermost range instead of using `range[range.Length - 1]` directly.
If you are trying to avoid calculating `range.Length - 1]` each time, you could do this:
```c#
var alpha = 0f
var rangeStep = 255.0f * i.Time / info.DisplayDuration / i.Range.Length;
...
	alpha += rangeStep;
```
 \- OR for some shading to outermost ellipse -
```c#
var rangeStep = 255.0f * i.Time / info.DisplayDuration / (i.Range.Length - 1);
var alpha = rangeStep;
...
	alpha += rangeStep;
```